### PR TITLE
X-Resolution

### PR DESCRIPTION
**Pull request by axtimhaus**
Draft for implementing x-resolution within each unit.

Main features:
- `DiskElement`s to subdivide other units
- subunits as generalization of disk elements and sequence members
- defaults to pull hook values from preceeding `in_profiles` and subunit's `out_profiles`

### CODE REVIEW

File: pyroll/core/roll/roll.py [hunk 1]
```diff
@@ -50,6 +50,9 @@ class Roll(HookHost):
     temperature = Hook[float]()
     """Mean temperature."""
 
+    surface_temperature = Hook[float]()
+    """Effective surface temperature."""
```
Comment (ChRen95) [inline pyroll/core/roll/roll.py:54]: Effective?

File: pyroll/core/roll_pass/hookimpls/roll_pass.py [hunk 2]
```diff
@@ -94,6 +96,11 @@ def rel_elongation(self: RollPass):
     return self.abs_elongation / self.in_profile.length
 
 
+@RollPass.strain
+def strain(self: RollPass):
+    return self.log_elongation
+
+
 @RollPass.strain_rate
 def strain_rate(self: RollPass):
```
Comment (ChRen95) [inline pyroll/core/roll_pass/hookimpls/roll_pass.py:105]: Should be altered to plugin since this is Hoff and Dahl.

**Review by ChRen95 — APPROVED** [review]

File: pyroll/core/roll/roll.py (hunk 1 above)
Comment (axtimhaus) [reply pyroll/core/roll/roll.py:54]: Means not a local value on a certain surface point.

File: pyroll/core/roll_pass/hookimpls/roll_pass.py (hunk 2 above)
Comment (axtimhaus) [reply pyroll/core/roll_pass/hookimpls/roll_pass.py:105]: Correct. I think we should sit together and go through every core hookimpl and decide.

**Pull request merged**